base/sparse/cholmod.jl

```diff
@@ -790,9 +838,37 @@ function sparse{Ti}(A::Sparse{Complex{Float64},Ti}) # Notice! Cannot be type sta
     end
     return convert(Hermitian{Complex{Float64},SparseMatrixCSC{Complex{Float64},Ti}}, A)
 end
-sparse(L::Factor)   = sparse(Sparse(L))
+function sparse(F::Factor)
+    s = unsafe_load(F.p)
+    if bool(s.is_ll)
+        L = Sparse(F)
+        A = sparse(L*L')
+    else
+        LD = sparse(F[:LD])
+        L, d = getLd!(LD)
+        A = scale(L, d)*L'
+    end
+    p = get_perm(F)
+    if p != [1:s.n;]
+        pinv = Array(Int, length(p))
+        for k = 1:length(p)
+            pinv[p[k]] = k
+        end
+        A = A[pinv,pinv]
```

Comment:
Sorry for not reading carefully enough. I can see that `FactorComponent` acts like the suggested `PivotedCholeskyFactor`. I'm still in doubt if it is a good idea to combine `L` and `P` in one type. It is already on the todo to make a permutation matrix type and we could support extracting `P` from a non-pivoted dense Cholesky where it would then be almost a noop in `Px=y`. Then code could still be generic for dense and sparse input.